thunder/transforms/quantization.py

```diff
@@ -294,3 +295,214 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+
+        return state_dict
+
+    def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilogue_trace, **kwargs):
+        tm = self.thunder_module
+        from thunder.core.trace import tracectx
+
+        checks = get_checks(prologue_trace)
+
+        # do we need this?
+        # prologue_proxy_map = {
+        #     get_param_bsym.output.name: dict(
+        # shape=self.lora_linear_states[model_weight_name]["loraA.shape"],
+        # dtype=thunder.dtypes.to_dtype(self.lora_linear_states[model_weight_name]["loraA.dtype"]),
+        #     )
+        #     for model_weight_name, (check_bsym, get_param_bsym) in checks.items()
+        #     if model_weight_name in self.lora_linear_states
+        # }
+        # here we switch the prologue_trace to a copy with new metadata
+        # prologue_trace = trace_with_replaced_proxy_metadata(prologue_trace, prologue_proxy_map)
+
+        checks = get_checks(prologue_trace)
+        proglogue_to_compute_outputs = prologue_trace.output[0]
+        output_idxes = {o.name: i for i, o in enumerate(proglogue_to_compute_outputs)}
+
+        lora_linear_proxies: dict[str, str] = {}  # proxy_name -> param_name
+        additional_proxies: dict[str, thunder.Proxy] = {}  # param_name.(absmax|code) -> Proxy
+
+        new_bsyms = []
+        new_compute_inputs = []
+        for n, qs in self.lora_linear_states.items():
+            param = tm.get_parameter(n)
+            n_loraA = f"{n}.loraA"
+            n_loraB = f"{n}.loraB"
+            n_linear = f"{n}"
+            check, get_param = checks[n]
+            lora_linear_proxies[get_param.output.name] = n
+            # check has args: tensor, shape, device, dtype, requires_grad
+            proxy, _, device, _, requires_grad = check.args
+
+            output_idx = output_idxes.get(get_param.output.name)
+            if output_idx is not None:
+                with tracectx(prologue_trace):
+                    # better way
+                    proxy_loraA = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_loraA",
+                        shape=qs["loraA.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["loraA.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    proxy_loraB = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_loraB",
+                        shape=qs["loraB.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["loraB.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraA, output=proxy_loraA))
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_loraB, output=proxy_loraB))
+
```

Comment:
I think I forgot this in quantization,  but it would be cool to add a shape, device, dtype check here.